### PR TITLE
Create docs subdirectory and Sphinx scaffolding

### PR DESCRIPTION
**Pull request by sydneyli**
Just ran `sphinx-quickstart docs` in the root directory.

Only plugin/extension I added was `autodoc`. Let me know if you'd prefer it set up differently.

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-mock; python_version < '3.3'
\ No newline at end of file
+mock; python_version < '3.3'
```
Comment (etingof) [inline test-requirements.txt]: I can't help but wonder what has changed here?

**Comment by etingof** [comment]
Ah, we should probable introduce something like `test-requirements.txt` listing devel/tesing dependencies e.g. Sphinx...

**Comment by etingof** [comment]
May be in upcoming/other commit, we'd probably want to link some of the newly introduced dicstrings to this docs infrastructure to see how it works. I suspect you have done that already elsewhere, though. ;-)

File: test-requirements.txt (hunk 1 above)
Comment (sydneyli) [reply test-requirements.txt]: Must be a newline ending or something similar...

**Comment by sydneyli** [comment]
> These .gitignore files in the commit - do we need them?

The `.gitignore` files are to make git keep the `_static` and `_templates` subdirectories (the sphinx build seems to expect that they exist).

> Should we hook up docs building to the CI job?

Yep, will do!

**Comment by sydneyli** [comment]
> Should we hook up docs building to the CI job?

I'm having a bit of trouble getting this to work on Py 2.6/3.2/3.3 for the time being-- I'll revert for the time being and create a new issue for this. @etingof does that sound okay?

> Ah, we should probable introduce something like test-requirements.txt listing devel/tesing dependencies e.g. Sphinx...

Right now it's just `test-requirements` and `requirements`-- should we rename `test-requirements` to `dev-requirements` and add Sphinx as a dependency there?

**Comment by etingof** [comment]
> > Should we hook up docs building to the CI job?
> 
> I'm having a bit of trouble getting this to work on Py 2.6/3.2/3.3 for the time being-- I'll revert for the time being and create a new issue for this. @etingof does that sound okay?

Sure! I am curious what exactly is not working? On the other hand, those Pythons are seriously obsolete. If the issue is not trivial, may be we could just avoid building docs on them.

> > Ah, we should probable introduce something like test-requirements.txt listing devel/tesing dependencies e.g. Sphinx...
> 
> Right now it's just `test-requirements` and `requirements`-- should we rename `test-requirements` to `dev-requirements` and add Sphinx as a dependency there?

I think `requirements.txt` is a de-facto (?) standard for listing runtime dependencies. I'd not rename this file into `dev-requirements.txt`.

**Comment by sydneyli** [comment]
> On the other hand, those Pythons are seriously obsolete. If the issue is not trivial, may be we could just avoid building docs on them.

Yep, I think all those Pythons are actually all deprecated! I'll also make an issue for dropping support through Py 3.3.

I think this is good to go-- like the linter, I'm just building docs on the most recent Python for the time being.

**Comment by etingof** [comment]
> Yep, I think all those Pythons are actually all deprecated! I'll also make an issue for dropping support through Py 3.3.

They are indeed very old and rarely (?) used. We should probably drop them altogether once we run into a reasonable limitation... My only worry is that there might be people out there still stuck at 2.6...

**Pull request merged**